rust_src/src/vector_macros.rs

```diff
@@ -0,0 +1,15 @@
+macro_rules! offset_of {
```

Comment:
I recall problems with offset_of in remacs: https://github.com/remacs/remacs/issues/1564. I don't think the situation has changed in terms of there being no ready solution yet, however I think we should still track it. It should not block your PR, just wanted to point it out.